src/main/java/baritone/pathing/movement/movements/MovementTraverse.java

```diff
@@ -72,28 +72,40 @@ public static double cost(CalculationContext context, int x, int y, int z, int d
         IBlockState pb1 = context.get(destX, y, destZ);
         IBlockState destOn = context.get(destX, y - 1, destZ);
         Block srcDown = context.getBlock(x, y - 1, z);
+        if ((srcDown == Blocks.MAGMA || destOn.getBlock() == Blocks.MAGMA) && !context.allowSneakOverMagma) {
+            return COST_INF;
+        }
         if (MovementHelper.canWalkOn(context.bsi, destX, y - 1, destZ, destOn)) {//this is a walk, not a bridge
-            double WC = WALK_ONE_BLOCK_COST;
+            double WC;
+            if (srcDown == Blocks.MAGMA) {
+                WC = SNEAK_ONE_BLOCK_COST / 2;
+            } else if (srcDown == Blocks.SOUL_SAND) {
+                WC = WALK_ONE_OVER_SOUL_SAND_COST / 2;
+            } else {
+                WC = WALK_ONE_BLOCK_COST / 2;
+            }
+            if (destOn.getBlock() == Blocks.MAGMA) {
+                WC += SNEAK_ONE_BLOCK_COST/2;
+            } else if (destOn.getBlock() == Blocks.SOUL_SAND) {
+                WC += WALK_ONE_OVER_SOUL_SAND_COST / 2;
+            } else {
+                WC += WALK_ONE_BLOCK_COST / 2;
+            }
```

Comment:
Why did you move this out of the `else` branch?
It's just useless calculations if there's water to swim in.